app/controllers/tasks_controller.rb

```diff
@@ -39,13 +40,16 @@ def create
   def edit; end
 
   def update
-   if @task.update(task_params)
-     flash[:success] = "「#{@task.title}」のタスクを修正しました"
-     redirect_to tasks_path
-   else
-     flash.now[:danger] = "更新に失敗しました"
-     render :edit
-   end
+    unless params[:task][:label_ids]
+      @task.tasklabels.delete_all
+    end
```

Comment:
こちら1行で実行しても良いかと思います。
```suggestion
    @task.tasklabels.delete_all unless params[:task][:label_ids]
```